packages/aws-cdk-lib/aws-logs/lib/data-protection-policy.ts

```diff
@@ -0,0 +1,375 @@
+import { Stack } from '../../core';
+import { Construct } from 'constructs';
+/**
+ * Represents a data protection policy in a log group.
+ */
+export class DataProtectionPolicy {
+
+  /**
+   * Name of the data protection policy
+   *
+   * @default - 'data-protection-policy-cdk'
+   */
+  public readonly name: string;
+
+  /**
+   * Description of the data protection policy
+   *
+   * @default - 'cdk generated data protection policy'
+   */
+  public readonly description: string;
+
+  /**
+   * Version of the data protection policy
+   */
+  public readonly version: string;
+
+  /**
+   * Statements within the data protection policy. Must contain one Audit and one Redact statement
+   */
+  public statement: object;
+
+  constructor(scope: Construct, props: DataProtectionPolicyProps) {
+    this.name = props.name || 'data-protection-policy-cdk';
```

Comment:
Instead of this can you generate the name using `Names.uniqueResourceName`.

Reply:
I tried using this methods but it appears to be for Constructs/Resources specifically. Given this is a property, I was unable to use it without errors.